### PR TITLE
Rework proxy creation in Ruby

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR allows you to create a proxy with simply `FooBarPrx.new(communicator, proxyString)` in Ruby.

It also updates the tests to:
 - use the new `new`
 - remove many extraneous uncheckedCast

Note that since Ruby doesn't support method overloading, we can't use new as an alias for uncheckedCast. So uncheckedCast stays.

### CODE REVIEW

File: ruby/test/Ice/operations/BatchOneways.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
 def batchOneways(p)
     bs1 = "\0" * (10 * 1024);
 
-    batch = Test::MyClassPrx::uncheckedCast(p.ice_batchOneway())
+    batch = p.ice_batchOneway()
```
Comment (bernardnormier) [inline ruby/test/Ice/operations/BatchOneways.rb:8]: Proxy factory methods in Ruby preserve types, but the tests didn't know it?

File: ruby/src/IceRuby/Proxy.cpp [hunk 2]
```diff
@@ -1150,9 +1150,13 @@ IceRuby_ObjectPrx_ice_staticId(VALUE /*self*/)
 }
 
 extern "C" VALUE
-IceRuby_ObjectPrx_new(int /*argc*/, VALUE* /*args*/, VALUE /*self*/)
+IceRuby_ObjectPrx_new(VALUE self, VALUE communicator, VALUE proxyString)
```
Comment (bernardnormier) [inline ruby/src/IceRuby/Proxy.cpp:1153]: The method was already provided, but not implemented!

**Review by externl — APPROVED** [review]
Looks good!

**Review by pepone — APPROVED** [review]

**Comment by pepone** [comment]
> we can't use new as an alias for uncheckedCast. So uncheckedCast stays.

We typically check the number of arguments and their types for such methods.

**Comment by bernardnormier** [comment]
> > we can't use new as an alias for uncheckedCast. So uncheckedCast stays.
> 
> We typically check the number of arguments and their types for such methods.

I'll look into it in a follow-up PR.

**Pull request merged**